transaction_decoder/transaction_decoder_test.py

```diff
@@ -0,0 +1,112 @@
+
+        assert (
+            metadata.sender
+            == "erd18w6yj09l9jwlpj5cjqq9eccfgulkympv7d4rj6vq4u49j8fpwzwsvx7e85"
+        )
+        assert (
+            metadata.receiver
+            == "erd1qqqqqqqqqqqqqpgqmua7hcd05yxypyj7sv7pffrquy9gf86s535qxct34s"
+        )
+        assert metadata.value == 1076977887712805212893260
+        assert metadata.function_name == "swap_lkmex_to_egld"
+        assert metadata.function_args == ["0b377f261c3c7191", ""]
+        if metadata.transfers:
+            assert metadata.transfers[0].value == 1076977887712805212893260
+            if metadata.transfers[0].properties:
+                assert metadata.transfers[0].properties.collection == "LKMEX-aab910"
+                assert (
+                    metadata.transfers[0].properties.identifier == "LKMEX-aab910-2fb4e9"
+                )
+
+    def test_multi_esdt_nft_transfer(self):
+        tx_to_decode = TransactionToDecode()
+        tx_to_decode.sender = (
+            "erd1lkrrrn3ws9sp854kdpzer9f77eglqpeet3e3k3uxvqxw9p3eq6xqxj43r9"
+        )
+        tx_to_decode.receiver = (
+            "erd1lkrrrn3ws9sp854kdpzer9f77eglqpeet3e3k3uxvqxw9p3eq6xqxj43r9"
+        )
+        tx_to_decode.value = "0"
+        tx_to_decode.data = "TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDBkZjNiZWJlMWFmYTEwYzQwOTI1ZTgzM2MxNGE0NjBlMTBhODQ5ZjUwYTQ2OEAwMkA0YzRiNGQ0NTU4MmQ2MTYxNjIzOTMxMzBAMmZlM2IwQDA5Yjk5YTZkYjMwMDI3ZTRmM2VjQDRjNGI0ZDQ1NTgyZDYxNjE2MjM5MzEzMEAzMTAyY2FAMDEyNjMwZTlhMjlmMmY5MzgxNDQ5MUA3Mzc3NjE3MDVmNmM2YjZkNjU3ODVmNzQ2ZjVmNjU2NzZjNjRAMGVkZTY0MzExYjhkMDFiNUA="
+
+        decoder = TransactionDecoder()
+        metadata = decoder.getTransactionMetadata(tx_to_decode)
+
+        assert (
+            metadata.sender
+            == "erd1lkrrrn3ws9sp854kdpzer9f77eglqpeet3e3k3uxvqxw9p3eq6xqxj43r9"
+        )
+        assert (
+            metadata.receiver
+            == "erd1qqqqqqqqqqqqqpgqmua7hcd05yxypyj7sv7pffrquy9gf86s535qxct34s"
+        )
+        assert metadata.value == 0
+        assert metadata.function_name == "swap_lkmex_to_egld"
+        assert metadata.function_args == [
+            "0ede64311b8d01b5",
+            "",
+        ]
+        if metadata.transfers:
+            assert len(metadata.transfers) == 2
+            assert metadata.transfers[0].value == 45925073746530627023852
+            if metadata.transfers[0].properties:
+                assert metadata.transfers[0].properties.collection == "LKMEX-aab910"
+                assert (
+                    metadata.transfers[0].properties.identifier == "LKMEX-aab910-2fe3b0"
+                )
+            assert metadata.transfers[1].value == 1389278024872597502641297
+            if metadata.transfers[1].properties:
+                assert metadata.transfers[1].properties.collection == "LKMEX-aab910"
+                assert (
+                    metadata.transfers[1].properties.identifier == "LKMEX-aab910-3102ca"
+                )
+
+    def test_esdt_transfer(self):
```

Comment:
could add a test for regular sc calls, such as:

```
def test_sc_call(self):
        tx_to_decode = TransactionToDecode()

        tx_to_decode.sender = (
            "erd1wcn58spj6rnsexugjq3p2fxxq4t3l3kt7np078zwkrxu70ul69fqvyjnq2"
        )
        tx_to_decode.receiver = (
            "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"
        )
        tx_to_decode.value = "0"
        tx_to_decode.data = (
            "d2l0aGRyYXdHbG9iYWxPZmZlckAwMTczZDA="
        )

        decoder = TransactionDecoder()
        metadata = decoder.getTransactionMetadata(tx_to_decode)

        assert (
            metadata.sender
            == "erd1wcn58spj6rnsexugjq3p2fxxq4t3l3kt7np078zwkrxu70ul69fqvyjnq2"
        )
        assert (
            metadata.receiver
            == "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"
        )
        assert (
            metadata.function_name
            == "withdrawGlobalOffer"
        )
        assert (
            metadata.function_args
            == ['0173d0']
        )
```

Reply:
Added the test.